src/Ledger/Utxo/Properties.lagda

```diff
@@ -4,37 +4,29 @@
 \begin{code}[hide]
 {-# OPTIONS --safe #-}
 
-open import Ledger.Transaction
-
-module Ledger.Utxo.Properties (txs : TransactionStructure) where
-
-open import Prelude
-open import Ledger.Prelude
-
-import Data.Nat as ℕ
 open import Algebra.Morphism            using (module MonoidMorphisms; IsMagmaHomomorphism)
+import Data.Nat as ℕ
 open import Data.Nat.Properties         hiding (_≟_)
 open import Data.Sign                   using (Sign)
 open import Data.Integer as ℤ           using (ℤ; _⊖_)
 open import Data.Integer.Ext            using (posPart; negPart)
 import Data.Integer.Properties as ℤ
 open import Relation.Binary             using (IsEquivalence)
+
+open import Prelude; open Equivalence
+
 open import Tactic.Cong                 using (cong!)
 open import Tactic.EquationalReasoning  using (module ≡-Reasoning)
 open import Tactic.MonoidSolver         using (solve-macro)
+open Tactic.EquationalReasoning.≡-Reasoning {A = ℕ} (solve-macro (quoteTerm +-0-monoid))
 
-open TransactionStructure txs
+open import Ledger.Prelude; open Properties; open Computational
+open import Ledger.Transaction
```

Comment:
I don't agree that it's better to keep imports local, I believe imports should be at the top of the file (as much as possible, since imports that depend on the current module parameters have to come after) so that dependencies can be identified easily, otherwise you would have files like so:
```
import Data...

<code>

import Relation...

<code>

lemma : ....
lemma = ....
  where
    open import Data...Properties 
    open import Relation....Properties
```

An alternative that is not that bad is to put only the necessary initial imports that the module parameters need, and then immediately afterwards import everything else, but I opted for the "as-top-as-possible" approach that I personally prefer.

Reply:
I see.  Re. the last paragraph, I agree, but I tend to favor having the `module` declaration as high as possible, so only the imports required for the module parameters come first, then the `module` declaration, than all the imports that can possibly come next (i.e., those that don't depend on definitions to come later).